src/commands/resetIndex.js

```diff
@@ -0,0 +1,65 @@
+import path from 'path'
+
+import { GitIndexManager } from '../managers/GitIndexManager.js'
+import { GitObjectManager } from '../managers/GitObjectManager.js'
+import { FileSystem } from '../models/FileSystem.js'
+
+import { readObject } from './readObject'
+import { log } from './log'
+
+/**
+ * Reset a file from the git index (aka staging area)
+ *
+ * @link https://isomorphic-git.github.io/docs/resetIndex.html
+ */
+export async function resetIndex ({
+  dir,
+  gitdir = path.join(dir, '.git'),
+  fs: _fs,
+  filepath,
+  ref = 'HEAD'
+}) {
+  try {
+    const fs = new FileSystem(_fs)
+    const tree = await log({ gitdir, fs, ref, depth: 1 })
+    let oid = tree[0] && tree[0].tree
+    let workdirOid
+    if (oid) {
+      try {
+        const obj = await readObject({ gitdir, fs, oid, filepath, format: 'deflated' })
+        oid = obj && obj.oid
+      } catch (e) {
+        oid = null
+      }
+      const object = await fs.read(path.join(dir, filepath))
+      workdirOid = await GitObjectManager.hash({
+        gitdir,
+        type: 'blob',
+        object
+      })
```

Comment:
Do wrap `workdirOid = ...` in a `if (object) { ... }` because object can be null if file is missing!

Reply:
agree

Reply:
is this will be `null` or will be an exception?